spring-integration-file/src/main/java/org/springframework/integration/file/remote/gateway/AbstractRemoteFileOutboundGateway.java

```diff
@@ -1351,12 +1387,39 @@ protected String getRemoteFilename(String remoteFilePath) {
 		}
 	}
 
+	private FileExistsMode resolveFileExistsMode(Message<?> message) {
+		if (this.fileExistsModeExpression != null) {
+			Object evaluationResult = this.fileExistsModeExpression.getValue(this.standardEvaluationContext, message);
+			if (evaluationResult == null) {
+				return this.fileExistsMode;
+			}
+			else if (evaluationResult instanceof FileExistsMode resolvedMode) {
+				return resolvedMode;
+			}
+			else if (evaluationResult instanceof String modeAsString) {
+				try {
+					return FileExistsMode.valueOf(modeAsString.toUpperCase());
+				}
+				catch (IllegalArgumentException ex) {
+					throw new MessagingException(message,
+							"Invalid FileExistsMode string: '" + modeAsString + "'. Expected one of: " +
+									Arrays.toString(FileExistsMode.values()), ex);
+				}
+			}
+			else {
+				throw new MessagingException(message,
+						"Expression returned invalid type for FileExistsMode: " +
+								evaluationResult.getClass().getName() + ". Expected FileExistsMode or String.");
+			}
+		}
+		return this.fileExistsMode;
+	}
+
 	private File generateLocalDirectory(Message<?> message, String remoteDirectory) {
-		EvaluationContext evaluationContext = ExpressionUtils.createStandardEvaluationContext(getBeanFactory());
 		if (remoteDirectory != null) {
-			evaluationContext.setVariable("remoteDirectory", remoteDirectory);
+			this.standardEvaluationContext.setVariable("remoteDirectory", remoteDirectory);
```

Comment:
This is not correct.
We cannot mutate a global context.
That's why the logic in this method uses its own local context instance.
We are going to mutate it with this specific variable.
The other thread has its own context instance, so it can mutate it with its own local variable.
when we do that on a global context, we make mistake.

Reply:
Actually, I haven't yet fully understood the usage pattern of EvaluationContext..
However, when following the logic, I determined that the return value of `ExpressionUtils.createStandardEvaluationContext(getBeanFactory())` retrieves the `integrationEvaluationContext` Bean registered in `DefaultConfiguringBeanFactoryPostProcessor`.
So I thought it would be the same instance as the global context set in `doInit()`.
I would greatly appreciate if you could clarify where my understanding might be incorrect on this matter.

Reply:
Right. Just because the `EvaluationContext` could be mutated, the bean producing it from the `BeanFactory` is a `prototype`. See `IntegrationEvaluationContextFactoryBean` and its `isSingleton()` implementation.
Therefore `ExpressionUtils.createStandardEvaluationContext()` always returns for us a fresh `EvaluationContext` instance. And since sometimes we don't need to mutate it, that is OK to have a global one for the whole component lifecycle.

Reply:
Thank you for explaining the `EvaluationContext` behavior! I understand now that `ExpressionUtils.createStandardEvaluationContext()` always returns a fresh instance due to the `prototype` scope of `IntegrationEvaluationContextFactoryBean`. This clarifies my confusion and will help me better understand Spring Integration going forward!